pkg/users/group-create-dialog.js

```diff
@@ -0,0 +1,170 @@
+
+import { has_errors } from "./dialog-utils.js";
+
+const _ = cockpit.gettext;
+
+function GroupCreateBody({ state, errors, change }) {
+    const {
+        name, id, user_specified_id_enabled,
+    } = state;
+
+    return (
+        <Form isHorizontal onSubmit={apply_modal_dialog}>
+            <FormGroup label={_("Name")}
+                       helperTextInvalid={errors && errors.name}
+                       validated={(errors && errors.name) ? "error" : "default"}
+                       fieldId="groups-create-name">
+                <TextInput id="groups-create-name"
+                           validated={(errors && errors.name) ? "error" : "default"}
+                           value={name} onChange={value => change("name", value)} />
+            </FormGroup>
+
+            <FormGroup label={_("ID")}
+                       hasNoPaddingTop
+                       helperTextInvalid={errors && errors.id}
+                       validated={(errors && errors.id) ? "error" : "default"}
+                       isStack
+                       fieldId="groups-create-id">
+                <Flex spaceItems={{ default: 'spaceItemsSm' }} alignItems={{ default: 'alignItemsCenter' }}>
+                    <Checkbox id="new-group-use-custom-id"
+                              label={_("Use specific ID")}
+                              isChecked={user_specified_id_enabled} onChange={checked => change("user_specified_id_enabled", checked)} />
+                    <FlexItem spacer={{ default: 'spacerLg' }}>
+                        <Popover bodyContent={_("When not specifying the ID the default is to use the smallest ID value greater than or equal to GID_MIN and greater than every other group.")}
+                                 showClose={false}>
+                            <HelpIcon />
+                        </Popover>
+                    </FlexItem>
+                </Flex>
+                {user_specified_id_enabled && <TextInput id="groups-create-id"
+                           validated={(errors && errors.id) ? "error" : "default"}
+                           value={id} onChange={value => change("id", value)} />}
+            </FormGroup>
+        </Form>
+    );
+}
+
+function is_valid_char_name(c) {
+    return (c >= 'a' && c <= 'z') ||
+        (c >= 'A' && c <= 'Z') ||
+        (c >= '0' && c <= '9') ||
+        c == '.' || c == '_' || c == '-';
+}
+
+function validate_name(name, groups) {
+    if (!name)
+        return _("No group name specified");
+
+    for (let i = 0; i < name.length; i++) {
+        if (!is_valid_char_name(name[i]))
+            return _("The group name can only consist of letters from a-z, digits, dots, dashes and underscores.");
+    }
+
+    for (let k = 0; k < groups.length; k++) {
+        if (groups[k].name == name)
+            return _("A group with this name already exists");
+    }
+
+    return null;
+}
+
+function validate_id(id) {
+    if (!id || isNaN(id))
+        return _("No valid id specified");
+}
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18136-20230109-094226-b26e04a5-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)